PSMDB-1905 Release notes 8.0.17-6 (#1091)
percona/psmdb-docs: docs/release_notes/8.0.17-6.md

title: Percona Server for MongoDB 8.0.17-6 ({{date.8_0_17}})
summary: Learn about improvements, new features, and bug and security vulnerability fixes in this release
authors:
    - Anastasia Alexandrova
version: 8.0.17-6
---

# Percona Server for MongoDB {{ page.meta.version }} ({{date.8_0_17}}) 

[Install](../install/index.md){.md-button}
[Upgrade from MongoDB Community](../install/upgrade-from-mongodb.md){.md-button}

Percona Server for MongoDB {{ page.meta.version }} is an enhanced, source-available, and highly-scalable database that is a
fully-compatible, drop-in replacement for MongoDB Community Edition.

Percona Server for MongoDB {{ page.meta.version }} includes the improvements and bug fixes of [MongoDB 8.0.17 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.17---dec-19--2025)

Percona Server for MongoDB {{ page.meta.version }} supports protocols and drivers of MongoDB Community 8.0.17.

> ## Release Highlights

Reviewer: Let's call this exceptionally 
```suggestion
## Security Update: Addressing "Mongobleed"
```

Author: Updated the title

> This release fixes security vulnerability [CVE-2025-14847: CWE-130](https://radar.offseq.com/threat/cve-2025-14847-cwe-130-improper-handling-of-length-dd0a4a45) in how MongoDB uses zlib compression library. Attackers with network access to `mongod` or `mongos` can extract fragments of uninitialized server memory without authentication if zlib compression is enabled. This memory may contain sensitive data, which poses a serious information disclosure risk. 

Reviewer: ```suggestion
This release fixes a security vulnerability [CVE-2025-14847: CWE-130](https://radar.offseq.com/threat/cve-2025-14847-cwe-130-improper-handling-of-length-dd0a4a45), a high-severity vulnerability colloquially known as "Mongobleed."

### What is Mongobleed?
The flaw stems from the improper handling of length headers within the zlib compression library integration. Much like the famous "Heartbleed" bug, this vulnerability allows a remote, unauthenticated attacker to "bleed" information from the server's memory.

### Why is it important? 
**The Mechanism**: If `zlib` compression is enabled, attackers with network access to `mongod` or `mongos` instances can send specially crafted packets.

**The Impact:** The server may respond by returning fragments of uninitialized memory.

**The Risk**: This leaked memory can contain sensitive data, including session tokens, internal metadata, or fragments of other active data sets, posing a significant risk of information disclosure.

**Recommendation**: We strongly recommend that all users upgrade to this version immediately, especially if your deployment utilizes zlib compression for client-to-server communication.
```

> This vulnerability affects the following versions of MongoDB Community Edition and Percona Server for MongoDB:

Reviewer: ```suggestion
### What Percona Server for MongoDB versions are affected?
This vulnerability affects the following versions of Percona Server for MongoDB:
```

> * MongoDB 8.0.0 through 8.0.16

Reviewer: ```suggestion
* Percona Server for MongoDB 8.0.0 through 8.0.16
```

> * MongoDB 7.0.0 through 7.0.26

Reviewer: ```suggestion
* Percona Server for MongoDB 7.0.0 through 7.0.26
```

> * MongoDB 6.0.0 through 6.0.26

Reviewer: ```suggestion
* Percona Server for MongoDB 6.0.0 through 6.0.26
```

> * MongoDB 5.0.0 through 5.0.31

Reviewer: ```suggestion
* Percona Server for MongoDB 5.0.0 through 5.0.31
```

> * MongoDB 4.4.0 through 4.4.29

Reviewer: ```suggestion
* Percona Server for MongoDB 4.4.0 through 4.4.29
```

> * All MongoDB Server v4.2 versions

Reviewer: ```suggestion
* All Percona Server for MongoDB Server v4.2 versions
```

> * All MongoDB Server v4.0 versions

Reviewer: ```suggestion
* All Percona Server for  MongoDB Server v4.0 versions
```

> * All MongoDB Server v3.6 versions

Reviewer: ```suggestion
* All Percona Server for MongoDB Server v3.6 versions
```

The issue is fixed upstream and is included in Percona Server for MongoDB 7.0.28-15 and 8.0.17-6. We strongly recommend upgrading to this latest minor version to ensure your deployments remain secure.

> If upgrading is not possible right away, disable `zlib` compression by configuring `networkMessageCompressors` (or `net.compression.compressors`) in configuration or startup parameters. Safe alternatives are `snappy`, `zstd` or `disabled`. 

Reviewer: ```suggestion
### How to apply a workaround?
If upgrading is not possible right away, disable `zlib` compression by configuring `networkMessageCompressors` (or `net.compression.compressors`) in configuration or startup parameters. Safe alternatives are `snappy`, `zstd` or `disabled`. 
```

=== "via Configuration File"

    To disable `zlib` compression, edit the `mongod.conf` or `mongos.conf` configuration file and set the `net.compression.compressors` parameter as follows:

    ```yaml
    net:
      compression:
        compressors: snappy,zstd,disabled
    ```

    Then restart the `mongod` or `mongos` service to apply the changes.

=== "via Command Line"

    To disable `zlib` compression, start `mongod` or `mongos` with the `--networkMessageCompressors` parameter as follows:

    ```bash
    mongod --networkMessageCompressors snappy,zstd
    ```

    Or for `mongos`:

    ```bash
    mongos --networkMessageCompressors snappy,zstd
    ```

Verify that `zlib` is disabled using the `db.adminCommand({ getCmdLineOpts: 1 })`. Ensure logs show no zlib negotiation when you try to connect to the server with a client that explicitly requests `zlib`.

> ### Upstream Improvements

Reviewer: I think we can remove this section - as it doesn't include anything additionall

Author: I updated it with the upstream Jira item so I believe we should leave it

Find the full list of changes in the release notes for:

> * [MongoDB 8.0.17 Community Edition](https://www.mongodb.com/docs/manual/release-notes/8.0/#8.0.17---dec-19--2025) 

Reviewer: Is that needed? It duplicates anyway the link mentioned in the beginning. I know that's how we did in the past.

Author: Removed
